src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -265,6 +266,13 @@ public bool Contains(T item)
                     {
                         return true;
                     }
+
+                    if (collisionCount >= _slots.Length)
```

Comment:
Take `_slots` to a local ref for its multiple uses?
`Slot[] slots = _slots;`

Reply:
just curious and to learn, what is the benefit of doing that?

Reply:
It uses the array in a register rather than a memory location; also applies extra optimisations because it cannot change between uses unlike the memory location version

Reply:
Makes sense. Thanks for explaining :)

Reply:
@benaadams is this something the JIT might be able to do itself in future? Always a shame to write code in an artificial way.

Reply:
Possibly, though it bounds checks have already regressed to become even more conservative when they come from memory https://github.com/dotnet/coreclr/pull/15756